interface/grpc/core/core.go

```diff
@@ -0,0 +1,57 @@
+package core
+
+import service "github.com/mesg-foundation/core/service"
+
+func toGRPCServices(ss []*service.Service) []*Service {
+	services := make([]*Service, 0)
+	for _, s := range ss {
+		services = append(services, toGRPCService(s))
+	}
+	return services
+}
+
+func toGRPCService(s *service.Service) *Service {
+	sv := &Service{
+		Id:          s.ID,
+		Name:        s.Name,
+		Description: s.Description,
+		Tasks:       map[string]*Task{},
+	}
+
+	for taskKey, task := range s.Tasks {
+		t := &Task{
+			Key:         task.Key,
+			Name:        task.Name,
+			Description: task.Description,
+			ServiceName: task.ServiceName,
+			Inputs:      toGRPCParameters(task.Inputs),
+			Outputs:     map[string]*Output{},
+		}
+		for outputKey, output := range task.Outputs {
+			t.Outputs[outputKey] = &Output{
+				Key:         output.Key,
+				Name:        output.Name,
+				Description: output.Description,
+				TaskKey:     output.TaskKey,
+				ServiceName: output.ServiceName,
+				Data:        toGRPCParameters(output.Data),
+			}
+		}
+		sv.Tasks[taskKey] = t
+	}
+
+	return sv
+}
+
+func toGRPCParameters(params map[string]*service.Parameter) map[string]*Parameter {
```

Comment:
should we use `toProtoParameter` ? I think the data are related to protobuf and sent via grpc. Small detail and I'm fine with that it's just to think about it maybe

Reply:
done